synapse/federation/federation_server.py

```diff
@@ -698,6 +698,19 @@ async def check_server_matches_acl(self, server_name: str, room_id: str):
 
         raise AuthError(code=403, msg="Server is banned from room")
 
+    async def on_user_info(self, user_ids: List[str]):
+        """Return user information given a list of user ids
+
+        Args:
+            user_ids: A list of users to return information about
+
+        Returns:
+            A dictionary mapping each user ID to a dict with the following keys:
+                * expired (bool) - whether this is an expired user
```

Comment:
Mm, yeah this would work in this instance (and you're right, this should definitely have a `-> Dict[str, bool]` on it), though in cases where you have a dictionary with values of different types it's useful to specify which is which.

Reply:
Added the return signature.

Reply:
Could you do that to the other functions that this comment would apply to please?

Reply:
Yeah, though be aware I can't do it for `@defer.inlineCallbacks` functions.

Reply:
> Yeah, though be aware I can't do it for `@defer.inlineCallbacks` functions.

Ugh, I didn't know about that. lgtm then.